test/webgpu_required_limits.c

```diff
@@ -0,0 +1,97 @@
+#include <assert.h>
+#include <emscripten.h>
+#include <stdio.h>
+#include <webgpu/webgpu.h>
+
+int adapter_limit_maxColorAttachmentBytesPerSample = -1;
+
+EM_ASYNC_JS(int, get_limit, (), {
+  if (!navigator.gpu) {
+    throw Error("WebGPU not supported.");
+  }
+
+  const adapter = await navigator.gpu.requestAdapter();
+  if (!adapter) {
+    throw Error("Couldn't request WebGPU adapter.");
+  }
+
+  console.log("adapter limits", adapter.limits);
+
+  return adapter.limits.maxColorAttachmentBytesPerSample;
+});
+
+void on_device_request_ended(WGPURequestDeviceStatus status,
+                             WGPUDevice device,
+                             char const* message,
+                             void* userdata) {
+  assert(status == WGPURequestDeviceStatus_Success);
+
+  WGPUSupportedLimits device_supported_limits;
+  wgpuDeviceGetLimits(device, &device_supported_limits);
+
+  printf("required maxColorAttachmentBytesPerSample=%d\n",
+         adapter_limit_maxColorAttachmentBytesPerSample);
+  printf("supported maxColorAttachmentBytesPerSample=%d\n",
+         device_supported_limits.limits.maxColorAttachmentBytesPerSample);
+
+  // device supported limit MUST be equal or larger than what was requested,
+  // otherwise device acquisition should have failed
+  assert(device_supported_limits.limits.maxColorAttachmentBytesPerSample >=
+         adapter_limit_maxColorAttachmentBytesPerSample);
```

Comment:
The device limits should be always exactly what was requested, so this can test `==`.